lib/init/grass.py

```diff
@@ -2243,15 +2012,12 @@ def get_username():
         user = os.getenv("USER")
         if not user:
             user = os.getenv("LOGNAME")
-        if not user:
+        if not user and (whoami_executable := shutil.which("whoami")):
```

Comment:
> Never saw the `:=` syntax in Python

walrus operator :=))) Seriously.

> but the shutil.which is great (and the commit message is great too)

Definitively, the subprocess package and Python itself changed a lot since `lib/init/grass.py` was written.

Reply:
I already knew that name, but not in Python. I checked it out, it was added in Python 3.8.